pkg/client/cache/reflector.go

```diff
@@ -337,16 +304,23 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			}
 			return nil
 		}
-		if err := r.watchHandler(w, &resourceVersion, resyncCh, stopCh); err != nil {
-			if err != errorResyncRequested && err != errorStopRequested {
+
+		if err := r.watchHandler(w, &resourceVersion, stopCh); err != nil {
+			if err != errorStopRequested {
 				glog.Warningf("%s: watch of %v ended with: %v", r.name, r.expectedType, err)
 			}
 			return nil
 		}
-		if r.canForceResyncNow() {
-			glog.V(4).Infof("%s: next resync planned for %#v, forcing now", r.name, r.nextResync)
-			return nil
-		}
+		/*
```

Comment:
>
From our testing it appears that both understandings are kind of correct here. If I initiate a watch on version 10 and the latest version is 15 then I will get an update for version 11, but that's it. I would then have to initiate a watch on 11 to get 12, and so on. This has been our experience with both the command line tools and modifying a test in kubernetes. So while there is an ability to get updates for all changes from the certain point, all the changes do not come across a single watch. You have to continually re-issue the watch with the last version you received until you finally catch up.

I would be very surprised if that's the case, but if it is, that's really really really bad and we should fix that ASAP.
@rrati - can you please file an issue for it and provide a way to reproduce it?
@smarterclayton @lavalamp @kubernetes/sig-api-machinery